libs/common/RespReadUtils.cs

```diff
@@ -378,7 +378,7 @@ public static bool ReadSignedLengthHeader(out int length, ref byte* ptr, byte* e
             }
 
             // Validate length
-            if (value > int.MaxValue)
+            if (value > int.MaxValue + (ulong)(negative ? 1 : 0)) // int.MinValue = -int.MaxValue + 1
```

Comment:
maybe put the negative check insde the first `if` so that we do not always pay for the negative check